R/060_utils.R

```diff
@@ -0,0 +1,348 @@
+#' paste3
+#'
+#' An internal fucntion that removes NAs from paste.
+#' Source: https://stackoverflow.com/users/1855677/42.
+#'
+#' @param ... objects to be pasted together.
+#' @param sep separating string.
+
+paste3 <- function(..., sep = ", ") {
+  L <- list(...)
+  L <- lapply(L, function(x) {
+    x[is.na(x)] <- ""
+    x
+  })
+  ret <- gsub(
+    paste0("(^", sep, "|", sep, "$)"), "",
+    gsub(
+      paste0(sep, sep), sep,
+      do.call(paste, c(L, list(sep = sep)))
+    )
+  )
+  is.na(ret) <- ret == ""
+  ret
+}
+
+#' Supply an expression to identify which header groups are hooked.
+#'
+#' @description
+#' This function is used with the `.hook_if` or `.hook_if_rev` arguments in the `locate_groups` function.
+#' It passes an expression to `dplyr::summarise` that identifies which header groups are hooked - for example swiched from N to NNW.
+#' See the the `locate_groups` documentation for more information and an example.
+#'
+#' @param ...  expression applied to a  identifies which header groups are hooked
+#'
+#' @name get_range_dfs
+#' @export
+
+hook <- function(...) {
```

Comment:
Because this is a user-facing function, I think it belongs elsewhere.  `utils.R` really for internal functions (so I ought to have called it `internals.R`!)

Reply:
Agreed. It probably belongs in get_header_groups.R.

Reply:
Have moved this function to same file as other locate functions. 3deac522d21b4fa32c6600615c4bad41d5525f10.